monitoring_data_pipeline/post_cohort_repeatative_script/monitoring_data_old_to_raw_schema_tables_update_script.py

```diff
@@ -181,6 +170,119 @@
         attempted_at = EXCLUDED.attempted_at;
 """)
 
+student_pre_recorded_query = text("""
+    WITH raw_student_cohort_data AS (
+        SELECT student_id, cohort_code
+        FROM raw.student_cohort
+    ),
+    student_details_data AS (
+        SELECT id,email FROM raw.student_details
+    ),
+    resource_data AS (
+        SELECT id AS resource_id, title, total_duration AS watchtime_in_min
+        FROM raw.resource
+    ),
+    raw_student_session_info AS (
+        SELECT
+            "Email" AS email,
+            "Session_Code" AS session_code,
+            "Duration_in_secs" AS watchtime_in_secs,
+            "watched_on" AS watched_on
+        FROM old.student_session_information
+        WHERE "Session_Code" LIKE 'VID%' 
+           
+    ),
+    student_pre_recorded_cte AS (
+        SELECT
+            sd.id::INT AS student_id,
+            r.resource_id::INT AS resource_id,
+            sc.cohort_code AS cohort_code,
+            ssi.watchtime_in_secs::INT AS watchtime_in_sec,
+            ssi.watched_on::DATE AS watched_at
+        FROM raw_student_session_info ssi
+        INNER JOIN student_details_data sd ON ssi.email = sd.email                    
+        INNER JOIN raw_student_cohort_data sc ON sd.id = sc.student_id
+        INNER JOIN resource_data r ON ssi.session_code = r.title
+    )
+    INSERT INTO raw.student_pre_recorded (
+        student_id, resource_id, cohort_code, watchtime_in_sec, watched_at
+    )
+    SELECT * FROM student_pre_recorded_cte
+    ON CONFLICT (student_id, resource_id)
+    DO UPDATE SET                          
+        watchtime_in_sec = EXCLUDED.watchtime_in_sec,
+        watched_at = EXCLUDED.watched_at;
+""")
+
+student_cohort_query = text("""
+    WITH student_cohort_sheet AS (
+        SELECT
+            cohort_code,
+            student_id,
+            student_role
+        FROM public.uploadsincubator_batch_data_20251208101928
+    ),
+    
+    student_details_data AS (
+        SELECT
+            id AS student_id,
+            email
+        FROM raw.student_details
+    ),
+
+    cohort_data AS (
+        SELECT
+            cohort_code,
+            cohort_name,
+            start_date
+        FROM raw.cohort
+    ),
+
+    student_cohort_data AS (
+        SELECT
+            RIGHT(CAST(EXTRACT(YEAR FROM c.start_date) AS TEXT), 2)
+            || c.cohort_code
+            || LPAD(CAST(sd.student_id AS TEXT), 7, '0') AS student_code,
+
+            sd.student_id::INT AS student_id,
+            c.cohort_code::VARCHAR(6) AS cohort_code,
+
```

Comment:
I feel a lot of queries are added multiple times as subqueries. Can we create separate functions for them as reusables. We can do that in a separate PR.